proposals/dictionary-expressions.md

```diff
@@ -402,36 +316,152 @@ Synthesized types are free to employ any strategy they want to implement the req
+2. Purely permissive.  All elements are added using the indexer.  Perhaps with compiler warnings if the exact same key is given the same constant value twice.
+3. Perhaps a hybrid model.  `.Add` if only using `k:v` and switching to indexers if using spread elements.  There is deep potential for confusion here.
+
+**Resolution:** Use *indexer* as the lowering form. [LDM-2024-03-11](https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-03-11.md#conclusions)
+
+## Retracted Designs/Questions
 
-There is a subtle concern around the following interface destinations:
+### Question: Should `k:v` elements force dictionary semantics?
+
+Is there a concern around the following interface destinations:
 
 ```c#
 // NOTE: These are not overloads.
 
-void Xxx(IEnumerable<KeyValuePair<string, int>> pairs) ...
-void Yyy(IDictionary<string, int> pairs) ...
+void AAA(IEnumerable<KeyValuePair<string, int>> pairs) ...
+void BBB(IDictionary<string, int> pairs) ...
 
-Xxx(["mads": 21, .. ldm]);
-Yyy(["mads": 21, .. ldm]);
+AAA(["mads": 21, .. ldm]);
+BBB(["mads": 21, .. ldm]);
 ```
 
 When the destination is an IEnumerable, we tend to think we're producing a sequence (so "mads" could show up twice).  However, the use of the `k:v` syntax more strongly indicates production of a dictionary-value.
 
 What should we do here when targeting `IEnumerable<...>` *and* using `k:v` elements? Produce an ordered sequence, with possibly duplicated values?  Or produce an unordered dictionary, with unique keys?
 
-Working group recommendation: `IEnumerable<KVP>` is not a dictionary type (as it lacks an indexer).  As such, it has sequential value semantics (and can include duplicates).  This would happen today anyways if someone did `[.. ldm]` and we do not think the presence of a `k:v` element changes how the semantics should work.
+Decision: `IEnumerable<KVP>` is not a dictionary type (as it lacks an indexer).  As such, it has sequential value semantics (and can include duplicates).  This would happen today anyways if someone did `[.. ldm]` and we do not think the presence of a `k:v` element changes how the semantics should work.
 
-## Random open questions
+This is also similar to how passing to an `IEnumerable<T>` would differ from passing to some *set* type with normal collection expressions.  The target type *intentionally* affects semantics, and there is no expectation that across very different target types that one would receive the same resultant values with the same behaviors.  We do not view *dictionary types* or *key value pair elements* as changing the calculus here.
 
-### Question: Parsing ambiguity
+### Question: Allow deconstructible types?
 
-Parsing ambiguity around: `[a ? [b] : c]`
+Should we take a very restrictive view of `KeyValuePair<,>`?  Specifically, should we allow only that exact type?  Or should we allow any types with an implicit conversion to that type?  For example:
```

Comment:
This idea made a lot of people uncomfortable.  And we do not need it at all for a viable and good dictionary story.  So explicitly retracting this from the design.  This can always be revisited later.